v6_carrier_py/wrapper.py

```diff
@@ -0,0 +1,87 @@
+import os
+import pickle
+from io import StringIO
+
+import pandas as pd
+from SPARQLWrapper import SPARQLWrapper, CSV
+from vantage6.client import serialization
+from vantage6.tools import docker_wrapper
+from vantage6.tools.data_format import DataFormat
+from vantage6.tools.dispatch_rpc import dispact_rpc
+from vantage6.tools.util import info
+
+SPARQL_RETURN_FORMAT = CSV
+
+
+def sparql_wrapper(module: str):
+    info(f"wrapper for {module}")
+
+    # read input from the mounted inputfile.
+    input_file = os.environ["INPUT_FILE"]
+    info(f"Reading input file {input_file}")
+
+    # TODO: _load_data handles input deserialization. It should be a public function
+    input_data = docker_wrapper._load_data(input_file)
+
+    query = input_data['query']
+
+    # all containers receive a token, however this is usually only
+    # used by the master method. But can be used by regular containers also
+    # for example to find out the node_id.
+    token_file = os.environ["TOKEN_FILE"]
+    info(f"Reading token file '{token_file}'")
+    with open(token_file) as fp:
+        token = fp.read().strip()
+
+    endpoint = os.environ["DATABASE_URI"]
+
+    # Workaround because the endpoint is automatically prefixed with the data folder. However this does not make
+    # sense for a sparql endpoint.
+    endpoint = _fix_endpoint(endpoint)
+
+    info(f"Using '{endpoint}' as triplestore endpoint")
+
+    data = query_triplestore(endpoint, query)
```

Comment:
You will most likely end up with a hard-to-interpret list of triples that you then need to convert to a usable format, I am not sure if a pandas dataframe is the best way to send it to the algorithm. The data is graph data and not spreadsheet data. So maybe pass a rdflib.Graph() object instead of a pd.DataFrame?

If you are doing the pandas dataframe so you can run the same algorithms with different wrappers, isn't that a bit wishful thinking? I.e. the fact that it is in a pandas dataframe doesn't mean it is meaningful spreadsheet data.

Reply:
I think you are misunderstanding the output of the query. We are querying graph data, yes, but the output doesn't have to be triples, a bit similar to how a sql query does not have to return all the rows in a table, just the ones you selected. 

Example query (simplified version from an [example from wikidata](https://www.wikidata.org/wiki/Wikidata:SPARQL_query_service/queries/examples#Goats), try it out if you like):

```
PREFIX skos: <http://www.w3.org/2004/02/skos/core#>
SELECT ?item ?itemLabel 
WHERE 
{
  # Item should be instance of goat (wd:Q2934)
  ?item wdt:P31 wd:Q2934.
  ?item  skos:prefLabel ?itemLabel
}
```

Reply:
Yes, you are completely right. I didn't think about in that way (my knowledge is very limited when it comes to SPARQL queries). After your demo this was more clear, and I wrote [this comment](https://github.com/CARRIER-project/vantage6-algorithms/pull/20#issuecomment-717331325).

I think the crux is that this way of thinking about RDF triples is now quite natural for you, but it clearly isn't for me, and might also not be for data scientists just wanting to run an algorithm. So I would recommend documenting the kind of queries that you expect, and provide this example.

Reply:
Nice, I like the example! :)